### DISCUSSION TITLE
Stripping Authorization header

### DISCUSSION THREAD

**Discussion by chuckwondo** [earthaccess-dev/earthaccess#513] [Q&A]
Is there any particular reason why `earthaccess.auth.SessionWithHeaderRedirection` overrides the `rebuild_auth` method instead of the `should_strip_auth` method? It seems that the latter would not only be more appropriate and simpler, but would also avoid skipping some netrc logic in the super `rebuild_auth` method.

**Comment by chuckwondo** [comment]
@betolink or @mfisher87, either of you know the answer to this?

**Reply by betolink** [reply]
Looks like `should_strip_auth` should work too.

**Reply by mfisher87** [reply]
I believe the auth header stripping code evolved from code in Earthdata Login documentation, so it's likely we just used that known solution instead of looking for better way to use the requests API. I see code very similar to this in many projects that started before earthaccess was available.

 I think there has also been a degree of fear of messing with this code because we'd need to be completely sure we're not leaking creds. If we wanted to test this it would require adding a new testing methodology, I think?
